src/circuits/circuits-util.tcc

```diff
@@ -1,5 +1,5 @@
-#ifndef __ZETH_CIRCUITS_UTILS_TCC__
-#define __ZETH_CIRCUITS_UTILS_TCC__
+#ifndef __ZETH_CIRCUITS_CIRCUITS_UTILS_TCC__
```

Comment:
Can you please rename the file to `circuits_utils.tcc` (likewise for the hpp), and fix the includes directives that will be broken after this renaming?